app/services/get-capacity-view.js

```diff
@@ -1,25 +1,85 @@
 const getBreadcrumbs = require('./get-breadcrumbs')
 const getWorkloadReports = require('./data/get-workload-report-views')
+const getCapacityBreakdown = require('./data/get-capacity-breakdown')
 const tableCreator = require('./helpers/table-creator')
 const getOrganisationUnit = require('./helpers/org-unit-finder')
+const organisationConstant = require('../constants/organisation-unit')
+const percentageCalculator = require('./helpers/percentage-calculator')
 
 module.exports = function (id, capacityDateRange, organisationLevel) {
   var organisationalUnitType = getOrganisationUnit('name', organisationLevel)
-  var result = {}
-  var workloadReportsPromise
-
   if (organisationalUnitType === undefined) {
     throw new Error(organisationLevel + ' should be offender-manager, region, team, ldu or hmpps')
   }
 
-  workloadReportsPromise = getWorkloadReports(id, capacityDateRange.capacityFromDate.toISOString(), capacityDateRange.capacityToDate.toISOString(), organisationLevel)
-
+  var result = {}
   result.breadcrumbs = getBreadcrumbs(id, organisationLevel)
 
-  return workloadReportsPromise.then(function (results) {
+  return getWorkloadReports(id, capacityDateRange.capacityFromDate.toISOString(), capacityDateRange.capacityToDate.toISOString(), organisationLevel)
+  .then(function (results) {
+    result.capacityBreakdown = []
     result.capacityTable = tableCreator.createCapacityTable(id, organisationalUnitType, capacityDateRange, results)
     result.title = result.breadcrumbs[0].title
     result.subTitle = organisationalUnitType.displayText
+
+    if (organisationalUnitType !== organisationConstant.OFFENDER_MANAGER) {
+      return getCapacityBreakdown(id, organisationLevel)
+      .then(function (memberWorkloadReports) {
+        result.capacityBreakdown = parseCapacityBreakdown(memberWorkloadReports, organisationLevel)
+        return result
+      })
+    }
     return result
   })
 }
+
+var parseCapacityBreakdown = function (workloadReports, organisationLevel) {
+  var capacityBreakdown = []
+
+  if (organisationLevel === organisationConstant.TEAM.name) {
+    workloadReports.forEach(function (workloadReport) {
+      capacityBreakdown.push(buildCapacityBreakdownEntry(workloadReport))
+    })
+  } else if (organisationLevel !== organisationConstant.OFFENDER_MANAGER.name) {
+    var organisationMap = new Map()
+
+    workloadReports.forEach(function (workloadReport) {
+      var valueToAdd
+
+      if (organisationMap.has(workloadReport.name)) {
+        valueToAdd = organisationMap.get(workloadReport.name)
+        valueToAdd.push(workloadReport)
+      } else {
+        valueToAdd = [workloadReport]
+      }
+      organisationMap.set(workloadReport.name, valueToAdd)
+    })
+
+    organisationMap.forEach(function (reports, orgName) {
```

Comment:
Is this returned result the key value pair from the map? Should it be the other way around?

Reply:
I also thought this, but it def comes through as value,key

Reply:
Strange